### ISSUE TITLE
Frequent, unpredictable crashes on macOS 10.12

### ISSUE THREAD

**Issue by Forcer78** [mpv-player/mpv#6191]
### mpv version and platform

mpv 0.29.1 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on Sat Oct  6 06:38:41 EDT 2018
ffmpeg library versions:
   libavutil       56.14.100
   libavcodec      58.18.100
   libavformat     58.12.100
   libswscale      5.1.100
   libavfilter     7.16.100
   libswresample   3.1.100
ffmpeg version: 4.0.2

Built through Homebrew.

### Reproduction steps

Play media.

### Expected behavior

The media plays without issue.

### Actual behavior

At some point during playback, even without user interaction, mpv will likely crash.

### Log file

Log file from mpv: http://0x0.st/sgHD.log
Last 3 system crash logs: http://0x0.st/sgHk.crash, http://0x0.st/sgHd.crash, http://0x0.st/sgHn.crash

sgHn.crash corresponds to sgHD.log. These three crashes occur in different code and with different exception codes. I don't know what's going on, unless this is a low-level problem with e.g. poor Swift support: I have a series of other, more repeatable mpv 0.29.1 crashes due to "illegal instruction: 4" that certainly look Swift-related.

mpv 0.28.2 did not crash in this manner.

**Comment by Akemi** [comment]
i am on the exact same macOS version as you and don't experience any problems (though slightly different mov version). it's hard to say what's going on since the crash reports are different and don't tell me much.

did you actually build it via homebrew or did you just get the bottle from it? for the former could you please provide a build log, in the case of the latter it might be the ominous swift3/SDK10.12 bug. which wouldn't be our issue. so far it only made problems with fullscreen switching (#5585, #6012) but could possibly cause other crashes too.

you could check how the mpv binary was build with `otool -l /PATH/TO/mpv` and check for this entry:
```
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.12
      sdk 10.12
```
if it says sdk 10.12, or older, it's very likely that there is some kind of swift bug involved. the way it was could also be a problem.

since you are on 10.12 try to get the newest xcode and build mpv with SDK10.13/swift4, so we could maybe exclude the above problem.

**Comment by Forcer78** [comment]
This is the concatenation of build logs 03.python3 and 04.python3: http://0x0.st/sgZt.python3

My build includes openal-soft and libarchive, with a single-line patch to work around #5759. Xcode and Swift are as up-to-date as 10.12 will allow.

otool output:

```
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.12
      sdk 10.13
```

Incidentally, the stolendata bundle looks to be built with the 10.12 SDK:

```
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.10
      sdk 10.12
```

It's unclear how to square the idea of the 10.12 SDK being buggy with the idea that the stolendata binary should be a working reference, unless I've misunderstood.

The issues you linked to (#5585, #6012) include descriptions that match the possibly-related "illegal instruction" crashes I mentioned in the OP, although I'm not sure my crashes involved fullscreen (I was attempting to play audio).

**Comment by Akemi** [comment]
@stolendata confirmed in his https://github.com/mpv-player/mpv/issues/6012#issuecomment-408404852 that he built it with swift4/sdk10.13, so no idea why it says sdk 10.12 there tbh. though for the issue linked, the new build fixed the problem i mentioned before.

like i said the problem mentioned is most likely also influencing other things, just that it was somewhat reproducible by toggling fullscreen. lets leave this issue aside though, it seems it's unrelated anyway since your build.log looks fine.

the crash log that says `Trace/BPT trap: 5` seems to indicate some kind of dylib problem, if google can be trusted.

also still try following:
- try stolendata's builds and see if it has the same problem
- try to run it in a debugger and get a backtrace
- try disabling videotoolbox just for testing
- try with `--no-config`
- try this PR #5892 if possible

a small crash course for using the debugger.
1. type `lldb mpv PATH_TO_FILE/URL` + hit ENTER
2. type `run` into lldb + hit ENTER
3. when the crash occurs, type `bt` + hit ENTER
4. type `bt full` + hit ENTER
5. paste me the backtrace.

**Comment by Akemi** [comment]
on my end, with 10.12.6, it says `sdk 10.13` on my mpv binary. so i really wonder how that difference happens.

**Comment by Forcer78** [comment]
Here are two backtraces of mpv run normally: http://0x0.st/sgbw.trace, http://0x0.st/sgbx.trace

The first is a hard crash. The second is an exception that I am unsure would hang outside of lldb.

I will test other variations after hearing back.

**Comment by Akemi** [comment]
sry, i am not much wiser with those two backtraces (yet).

**Comment by Forcer78** [comment]
I rebuilt mpv last week with `--disable-swift`. I haven't seen the specific crashes described in the first post since then, though I did just get an apparently different one: https://0x0.st/sDjb.log

Given this, I don't think Swift is mature or stable enough to transition major parts of mpv to at this time.

**Comment by Akemi** [comment]
most likely fixed with #6703. since we removed swift 3 support. if it still happens with git master let me know.
